src/eth/Cdp.js

```diff
@@ -35,13 +37,29 @@ export default class Cdp {
       .get('web3')
       .currentAccount();
 
-    const getId = new Promise(resolve => {
-      tubContract.onlognewcup = function(address, cdpIdBytes32) {
-        if (currentAccount.toLowerCase() == address.toLowerCase()) {
-          this.removeListener();
-          resolve(ethersUtils.bigNumberify(cdpIdBytes32).toNumber());
-        }
-      };
+    const getId = new Promise(async resolve => {
+      let id;
+      if (this._web3Service.usingWebsockets()) {
+        const saiTub = this._smartContractService._getContractInfo(
+          contracts.SAI_TUB
+        );
+        id = (await this._web3Service.waitForMatchingEvent(
+          saiTub,
+          'LogNewCup',
+          log => {
+            return log.lad.toLowerCase() === currentAccount.toLowerCase();
+          }
```

Comment:
Is it also possible to filter on the transaction Id?  It's possible for an address to own more than 1 CDP, so just filtering on the cdp owner's address doesn't guarantee that we get the right CDP Id.

Per a short discussion [here]([https://github.com/makerdao/dai-plugin-governance/pull/6#issuecomment-444569541]), I was thinking about how to consolidate the way we get logs from a transaction.

Personally, I'm partial to the way its done in OasisOrder (and how I'm doing it in the gov plugin with ApproveLinkTransaction) - that is, getting the logs from the transaction receipt, which we get in TransactionObject anyway.  The benefit is that you know that the log came from the right transaction, and there's no need to deal with creating a subscription (only to unsubscribe from it right away).

The downside is that it might be a complicated refactor to cdp.js - its not immediately clear to me how it would done.  What do other people think?  Sorry for not mentioning this earlier.

Reply:
hmm. i had always assumed that it was done this way when creating a CDP was because no less convoluted way was available. (And we carried the pattern into ProxyCdp as well.) If just looking at `tx.receipt.logs` works, then we should definitely just do that...

Reply:
> Is it also possible to filter on the transaction Id? It's possible for an address to own more than 1 CDP, so just filtering on the cdp owner's address doesn't guarantee that we get the right CDP Id.

The only risk here is that we get the wrong ID if the same address is creating two CDPs at the same time, right? Not a huge issue.

Reply:
True, it's only an issue if two are opened at same time